types/wlr_surface.c

```diff
@@ -491,6 +491,8 @@ static void surface_commit_pending(struct wlr_surface *surface) {
 		surface->role->precommit(surface);
 	}
 
+	wlr_signal_emit_safe(&surface->events.precommit, surface->pending);
```

Comment:
"precommit" is confusing: it's not always triggered right before commit, it may be triggered right before the state is cached instead.

Reply:
Can be renamed to `prepare`.